File: src/Python/easy/string/problem1941.py
class Solution:
    def areOccurrencesEqual(self, s: str) -> bool:
        lettersObj = {}
        lettersSet = set()
        for i in range(len(s)):
            if s[i] in lettersObj:
                lettersObj[s[i]] += 1
            else:
                lettersObj[s[i]] = 1 
        for k in lettersObj:
            lettersSet.add(lettersObj[k])
        return True if len(lettersSet) == 1 else False



s = "abacbc"

solution = Solution()
solution.areOccurrencesEqual(s)

File: src/JavaScript/easy/string/problem1941.js
/**
 * @param {string} s
 * @return {boolean}
 */

// First solution

var areOccurrencesEqual = function(s) {
    for(let i = 0; i < s.length; i++){
        if(s.match(new RegExp(`${s[i]}`, 'g')).length * new Set([...s]).size != s.length){
            return false
        }
    }
    return true
};



// Second solution

var areOccurrencesEqual = function(s) {
    let objLetters = {}
    let lettersSet  =new Set()
    for(let i = 0; i < s.length; i++){
        objLetters?.[s[i]] ? objLetters[s[i]]++ : objLetters[s[i]] = 1
    }
    for(let k in objLetters){
        lettersSet.add(objLetters[k])
    }
    return lettersSet.size == 1 ? true : false
};


// let s = "abacbc"
let s = "aaabb"
// let s = "vvvvvvvvvvvvvvvvvvv"
// let s = "rmvlshxoylylmnejinoikmdbxqhebkxhxrrmnjhvyptbimlndqqvlessktortxjsenspyjkivqidebojdppbqrhdtvytkop"

areOccurrencesEqual(s)